feedback/extensions/filters.py

```diff
@@ -0,0 +1,153 @@
+"""
+Open edX Filters needed for instructor dashboard integration.
+"""
+import pkg_resources
+from crum import get_current_request
+from django.conf import settings
+from django.template import Context, Template
+from openedx_filters import PipelineStep
+from web_fragments.fragment import Fragment
+
+from lms.djangoapps.courseware.block_render import load_single_xblock, get_block_by_usage_id  # pylint: disable=import-error
+from xmodule.modulestore.django import modulestore  # pylint: disable=import-error
+from common.djangoapps.student.models import CourseEnrollment  # pylint: disable=import-error
+
+TEMPLATE_ABSOLUTE_PATH = "/instructor_dashboard/"
+BLOCK_CATEGORY = "feedback_instructor"
+
+
+class AddFeedbackTab(PipelineStep):
+    """Add forum_notifier tab to instructor dashboard."""
+
+    def run_filter(
+        self, context, template_name
+    ):  # pylint: disable=unused-argument, arguments-differ
+        """Execute filter that modifies the instructor dashboard context.
+        Args:
+            context (dict): the context for the instructor dashboard.
+            _ (str): instructor dashboard template name.
+        """
+        if not settings.FEATURES.get("ENABLE_FEEDBACK_INSTRUCTOR_VIEW", False):
+            return {
+                "context": context,
+            }
+
+        course = context["course"]
+        template = Template(self.resource_string(f"static/html/{BLOCK_CATEGORY}.html"))
+
+        request = get_current_request()
+
+        context.update(
+            {
+                "blocks": load_blocks(request, course),
+            }
+        )
+
+        html = template.render(Context(context))
+        frag = Fragment(html)
+        frag.add_css(self.resource_string(f"static/css/{BLOCK_CATEGORY}.css"))
+        frag.add_javascript(self.resource_string(f"static/js/src/{BLOCK_CATEGORY}.js"))
+
+        section_data = {
+            "fragment": frag,
+            "section_key": BLOCK_CATEGORY,
+            "section_display_name": "Course Feedback",
+            "course_id": str(course.id),
+            "template_path_prefix": TEMPLATE_ABSOLUTE_PATH,
+        }
+        context["sections"].append(section_data)
+
+        return {"context": context}
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string("feedback", path)
+        return data.decode("utf8")
+
+
+def load_blocks(request, course):
+    """
+    Load feedback blocks for a given course for all enrolled students.
+
+    Arguments:
+        request (HttpRequest): Django request object.
+        course (CourseLocator): Course locator object.
+    """
+    course_id = str(course.id)
+
+    feedback_blocks = modulestore().get_items(
+        course.id, qualifiers={"category": "feedback"}
+    )
+
+    blocks = []
+    students = CourseEnrollment.objects.filter(course_id=course_id).values_list(
+        "user_id", "user__username"
+    )
```

Comment:
can we use this instead? https://github.com/openedx/edx-platform/blob/ddabba458bf304e7ddca0507dfed57b66dbc1eef/openedx/core/djangoapps/enrollments/data.py#L102-L115

Reply:
fixed